apps/crud/src/app/services/todo.service.ts

```diff
@@ -0,0 +1,170 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { TodoItem } from '../models/todo.model';
+import { TodoApiService } from './todo-api.service';
+import {
+  catchError,
+  concatMap,
+  finalize,
+  map,
+  switchMap,
+  tap,
+} from 'rxjs/operators';
+import { Observable, of } from 'rxjs';
+
+interface TodoStoreState {
+  todos: TodoItem[];
+  isLoading: boolean;
+}
+
+const initialState: TodoStoreState = {
+  todos: [],
+  isLoading: false,
+};
+
+@Injectable()
+export class TodoService extends ComponentStore<TodoStoreState> {
+  private $todos = this.select((state) => state.todos);
+  private $isLoading = this.select((state) => state.isLoading);
+
+  public viewModel$ = this.select(
+    this.$todos,
+    this.$isLoading,
+    (todos, isLoading) => {
+      return {
+        todos,
+        isLoading,
+      };
+    }
+  );
```

Comment:
there is a new API where you can directly write : 
```ts
  public viewModel$ = this.select({
    todos$: this.$todos,
    isLoading$: this.$isLoading,
    });
```
And I like to add the debounce property to true on the viewModel only to update it only once.